### PR TITLE
Fixes #68 Handle exceptions of docs creation and generation

### LINKED ISSUE #68: Docs are published to gh-pages even after a failed documentation generation
### This is a
- [x] Bug report
- [ ] Feature request

### Current behaviour

I tried to test yaydoc. Apparently, there was a bug in it which lead to a failure in documentation generation. Even after an exception was raised in the travis logs, the script continued its execution and pushed the erroneous docs.
### Expected behaviour

The execution of the script should stopped in such a case.

### Travis log
Link: https://travis-ci.org/imujjwal96/prelimQuiz/jobs/238663584

#### Snapshot
image

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

Error checking is done on the following commands
1. Creation of source files,
2. Generation of documentations.

## Related Issue

Issue: #68 

## Motivation and Context

Even after an exception was raised in the travis logs, the script continued its execution and pushed the erroneous docs. This has to be rectified.
## How Has This Been Tested?

Made the changes in my test repository and generated docs with it.
Travis log: https://travis-ci.org/imujjwal96/prelimQuiz/jobs/238708224
## Screenshots (if appropriate):
image
## Types of changes

- [x] Bug fix 
- [ ] New feature

## Checklist:

- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y Please review.

File: generate.sh [hunk 1]
```diff
@@ -29,6 +29,10 @@ pip install -q -r requirements.txt
 
 # Setting up documentation sources
 sphinx-quickstart --ext-githubpages -q -v $VERSION -a $AUTHOR -p $PROJECTNAME -t templates/ -d html_theme=${DOCTHEME:-alabaster} -d html_logo=${LOGO:-} > /dev/null
+if [ $? -ne 0 ]; then
+  echo -e "Failed to setup source directory.\n"
```
Comment (pri22296) [inline generate.sh]: could we have a better error message. *source* doesn't seem correct here.

Comment (imujjwal96) [inline generate.sh]: Always open for suggestion. 
I just interpreted it from the description of `sphinx-quickstart`

Comment (pri22296) [inline generate.sh]: I meant *source* is usually associated with code. Maybe use something along the lines of **Failed to Initialize build directory**?

Comment (imujjwal96) [inline generate.sh]: What about, **Failed to initialize build process**

Comment (pri22296) [inline generate.sh]: sounds good.

**Pull request merged**